### PR TITLE
Render temp hp above health bar

### PR DESCRIPTION
**Pull request by CarlosFdez**
image

### CODE REVIEW

File: src/module/canvas/token/object.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { TokenDocumentPF2e } from "@module/scene";
+import { Graphics } from "@pixi/graphics";
```
Comment (stwlam) [inline src/module/canvas/token/object.ts:2]:
```suggestion
```

File: src/module/canvas/token/object.ts [hunk 2]
```diff
@@ -123,6 +124,56 @@ class TokenPF2e extends Token<TokenDocumentPF2e> {
         return flankingBuddies.some((b) => onOppositeSides(this, b, flankee));
     }
 
+    protected override _drawBar(number: number, bar: Graphics, data: TokenResourceData): void {
```
Comment (stwlam) [inline src/module/canvas/token/object.ts:127]:
```suggestion
    protected override _drawBar(number: number, bar: PIXI.Graphics, data: TokenResourceData): void {
```

**Review by stwlam — CHANGES_REQUESTED** [review]

File: src/module/canvas/token/object.ts [hunk 3]
```diff
@@ -123,6 +124,58 @@ class TokenPF2e extends Token<TokenDocumentPF2e> {
...
+            bar.beginFill(color, 1.0).drawRoundedRect(0, healthBarY, healthPercent * this.w, barHeight, 2);
+
+            // Draw the container (outermost border)
+            bar.beginFill(black, 0).lineStyle(bs, black, 1.0).drawRoundedRect(0, 0, this.w, h, 3);
+
+            // Set position
+            bar.position.set(0, number === 0 ? this.h - h : 0);
+        } else {
+            super._drawBar(number, bar, data);
+        }
```
Comment (stwlam) [inline src/module/canvas/token/object.ts:176]: Can this be checked up top and returned early?

Comment (CarlosFdez) [reply src/module/canvas/token/object.ts:176]: That'll make it trickier to extend later, but I guess it can be reorganized when it is.

Comment (CarlosFdez) [reply src/module/canvas/token/object.ts:176]: I'll push it, but I think it made it trickier to read :(

File: src/module/canvas/token/object.ts [hunk 4]
```diff
@@ -123,6 +123,62 @@ class TokenPF2e extends Token<TokenDocumentPF2e> {
...
+    /** Overrides _drawBar() to also draw pf2e variants of normal resource bars (such as temp health) */
+    protected override _drawBar(number: number, bar: PIXI.Graphics, data: TokenResourceData): void {
+        if (!canvas.dimensions) return;
+
+        const actor = this.document.actor;
+        const isHealth = data.attribute === "attributes.hp" && actor?.attributes.hp;
+
+        if (!isHealth) {
+            super._drawBar(number, bar, data);
+            return;
```
Comment (stwlam) [inline src/module/canvas/token/object.ts:135]:
```suggestion
            return super._drawBar(number, bar, data);
```

Comment (CarlosFdez) [reply src/module/canvas/token/object.ts:135]: Surprised you're ok with return void. I know core does it, but it always felt like a bad idea. That might be my previous programming exp talking.

**Review by stwlam — APPROVED** [review]

**Pull request merged**